### PR TITLE
Migrate virtualize library

### PR DESCRIPTION
**Pull request by jakzaizzat**
**Changes**

- [x] Migrate feed (trending/latest/user activity) to [tanstack virtual](https://tanstack.com/virtual/v3)
- [x] Migrate user gallery
- [x] Migrate edtior sidebar  
- [x] Added infinite scroll to feed
- [x] Fix blank feed item on re-route

### CODE REVIEW

File: apps/web/src/components/Feed/Socialize/CommentBox/CommentBoxIcon.tsx [hunk 1]
```diff
@@ -110,29 +109,27 @@ export function CommentBoxIcon({ queryRef, eventRef }: Props) {
 
       <AnimatePresence>
         {showCommentBox && (
-          <FloatingOverlay style={{ zIndex: 10 }}>
```
Comment (tbezman) [inline apps/web/src/components/Feed/Socialize/CommentBox/CommentBoxIcon.tsx:113]: Why did we get rid of the Floating overlay?

File: apps/web/src/components/Feed/FeedList.tsx [hunk 2]
```diff
@@ -56,100 +46,20 @@ export default function FeedList({
...
-          )}
-        </CellMeasurer>
-      );
-    },
-    [feedData, feedMode, handlePotentialLayoutShift, isRowLoaded, measurerCache, query]
-  );
+  const feeds = virtualizer.getVirtualItems();
+  useLayoutEffect(() => {
+    parentOffsetRef.current = parentRef.current?.offsetTop ?? 0;
+  }, []);
```
Comment (tbezman) [inline apps/web/src/components/Feed/FeedList.tsx:62]: Can you explain what this is doing? I tried playing around with it locally to get a feel for it but no matter what I change `scrollMargin` to, nothing changes on my screen. This feels ugly and I'm hoping we don't need it 😂

File: apps/web/src/components/Feed/FeedList.tsx [hunk 3]
```diff
@@ -159,73 +69,91 @@ export default function FeedList({
...
+    const [lastItem] = [...feeds].reverse();
+
+    if (!lastItem) {
+      return;
+    }
+
+    if (lastItem.index >= feedData.length - 1 && hasNext && !isLoading) {
+      handleLoadMoreClick();
+    }
+  }, [feedData.length, feeds, hasNext, isLoading, handleLoadMoreClick]);
```
Comment (tbezman) [inline apps/web/src/components/Feed/FeedList.tsx:82]: Pretty sure we don't want infinite scroll for the feed right?

File: apps/web/src/components/Feed/FeedList.tsx [hunk 4]
```diff
@@ -159,73 +69,91 @@ export default function FeedList({
...
-                // By default, react-virtualized's list has the css property `will-change` set to `transform`
-                // An element with `position: fixed` beneath an element with `will-change: transform` will
-                // be incredibly busted. You can read more about that [here](https://stackoverflow.com/questions/28157125/why-does-transform-break-position-fixed)
-                //
-                // Simply setting this back to it's original `auto` seems to do the trick and shouldn't have
-                // any serious performance implications from some trivial testing that was done.
-                style={{ willChange: 'auto', outline: 'none' }}
+    <StyledFeedList ref={parentRef} height={virtualizer.getTotalSize()}>
+      <StyledFeedListContainer
+        yPosition={(feeds[0]?.start ?? 0) - virtualizer.options.scrollMargin}
```
Comment (tbezman) [inline apps/web/src/components/Feed/FeedList.tsx:91]: feels weird man

File: apps/web/src/components/Feed/FeedList.tsx (hunk 2 above)
Comment (Robinnnnn) [reply apps/web/src/components/Feed/FeedList.tsx:62]: +1 would love to have comments in the code for stuff like this!

File: apps/web/src/components/Feed/FeedList.tsx (hunk 3 above)
Comment (Robinnnnn) [reply apps/web/src/components/Feed/FeedList.tsx:82]: we actually do now @tbezman !

File: apps/web/src/components/Feed/FeedList.tsx (hunk 4 above)
Comment (Robinnnnn) [reply apps/web/src/components/Feed/FeedList.tsx:91]: comment would be helpful

File: apps/web/src/components/Feed/Socialize/CommentBox/CommentBoxIcon.tsx (hunk 1 above)
Comment (jakzaizzat) [reply apps/web/src/components/Feed/Socialize/CommentBox/CommentBoxIcon.tsx:113]: It makes the screen flickering

https://user-images.githubusercontent.com/4480258/226788490-f225141d-dc97-43db-bb39-1009edf8dc8b.mp4

File: apps/web/src/components/Feed/FeedList.tsx (hunk 2 above)
Comment (jakzaizzat) [reply apps/web/src/components/Feed/FeedList.tsx]: I removed it 😅 . Seems like it doesnt do any changes. lol

File: apps/web/src/components/Feed/FeedList.tsx (hunk 4 above)
Comment (jakzaizzat) [reply apps/web/src/components/Feed/FeedList.tsx]: Added comment here!

**Review by Robinnnnn — APPROVED** [review]

**Pull request merged**